src/all_optimizations.py

```diff
@@ -956,29 +972,39 @@ def dlc(DATA_DIR, dlc_thresh):
     parser.add_argument('--dlc_thresh', type=float, default=0.8, help='The likelihood of the dlc points below which will be excluded from the optimization')
     parser.add_argument('--plot', action='store_true', help='Showing plots')
     args = parser.parse_args()
-    
-    ROOT_DATA_DIR = os.path.join("..", "data")
-    DATA_DIR = os.path.join(ROOT_DATA_DIR, os.path.normpath(args.data_dir))
-    
+
+    # ROOT_DATA_DIR = os.path.join("..", "data")
+    DATA_DIR = os.path.normpath(args.data_dir)
+    assert os.path.exists(DATA_DIR)
+    DLC_DIR = os.path.join(DATA_DIR, 'dlc')
+    assert os.path.exists(DLC_DIR)
+
+    # load DLC info
+    res, fps, tot_frames, _ = app.get_vid_info(DATA_DIR) # path to original videos
+    assert args.end_frame <= tot_frames, f'end_frame must be less than or equal to {tot_frames}'
+    end_frame = tot_frames if args.end_frame == -1 else args.end_frame
+
```

Comment:
Even though these changes reduce the repeated use of this block of code, for now, I feel like this code should stay where it was originally ie in the fte, ekf, ... functions so that the overall code in each of these functions is as similar as possible to their corresponding .ipynb, as it means less maintenance hassles for us if something gets changed somewhere. Let me know what you think